docs/argo-server-auth-mode.md

```diff
@@ -1,15 +1,15 @@
 # Argo Server Auth Mode
 
-You can choose which kube config the Argo Server uses:
+You can configure how the Argo Server authenticates to Kubernetes:
 
-* `server` - in hosted mode, use the kube config of service account, in local mode, use your local kube config.
-* `client` - requires clients to provide their Kubernetes bearer token and use that.
-* [`sso`](./argo-server-sso.md) - since v2.9, use single sign-on, this will use the same service account as per "server" for RBAC. We expect to change this in the future so that the OAuth claims are mapped to service accounts.
+* `server`: In [hosted mode](argo-server.md#hosted-mode), use the Server's Service Account. In [local mode](argo-server.md#local-mode), use your local kube config.
+* `client`: Use the Kubernetes [bearer token of clients](access-token.md).
+* `sso`: Use [single sign-on](argo-server-sso.md). This will use the same SA as `server` for RBAC, unless you have enabled [SSO RBAC](argo-server-sso.md#sso-rbac)
```

Comment:
Yea I noticed that at some point. It is entirely undocumented and not necessary with multiple auth mode flags, so I'd prefer _not_ to document it. Both to not add confusion for users and so that users do not rely on it if we remove it eventually.

Reply:
We should remove it in 4.0 then.

Reply:
Agreed, but rather in 3.6 or another minor